### PR TITLE
Move profile view permission to private only

### PR DESCRIPTION
**Pull request by Siraphop4Nene**

### CODE REVIEW

**Review by pontakornth — APPROVED** [review]
Look good to me.

**Review by SainTurDaY27 — APPROVED** [review]
Good

File: users/views.py [hunk 1]
```diff
@@ -49,19 +50,23 @@ def settings(request):
     })
 
 
-def profile(request, user_id):
+# all user -> login first
+# all use -> use
```
Comment (HelloYeew) [inline users/views.py]: Delete this note

File: users/views.py [hunk 2]
```diff
@@ -74,11 +79,36 @@ def profile(request, user_id):
...
+def profile_with_id(request, user_id):
+    """
+    Show a user's profile.
+    """
+    try:
+        user = NewProfile.objects.get(id=user_id)
+    except NewProfile.DoesNotExist:
+        messages.error(request, 'This user does not exist.')
+        return redirect('homepage')
+    if request.user.is_authenticated and (request.user.is_superuser or request.user.is_staff):
```
Comment (HelloYeew) [inline users/views.py]: `login_required` annotation already work as a middleware on `is_authenticate`

**Review by HelloYeew — CHANGES_REQUESTED** [review]

**Review by HelloYeew — APPROVED** [review]

**Pull request merged**